frontend/ui/widget/menu.lua

```diff
@@ -1176,7 +1185,71 @@ function Menu:getPageNumber(item_number)
     if #self.item_table == 0 or item_number == 0 then
         return 1
     end
-    return math.ceil(math.min(item_number, #self.item_table) / self.perpage)
+    if self.items_max_lines then
+        for page, items in ipairs(self.page_items) do
+            if item_number <= items[#items] then
+                return page
+            end
+        end
+        return #self.page_items
+    else
+        return math.ceil(math.min(item_number, #self.item_table) / self.perpage)
+    end
+end
+
+function Menu:setupItemHeights()
+    if #self.item_table == 0 then
+        self.page_items = {{}}
+        return
+    end
+
+    local face = Font:getFace("smallinfofont", self.font_size)
+    local line_height = TextBoxWidget:new{
+        text = "A",
+        face = face,
+    }:getSize().h
+    local infont_size = self.items_mandatory_font_size or (self.font_size - 4)
+    local infont_face = Font:getFace("infont", infont_size)
+    local infont_char_width = TextWidget:new{
+        text = "0",
+        face = infont_face,
+        bold = true,
+    }:getSize().w
+    local available_width = self.inner_dimen.w
+    if self.is_enable_shortcut then
+        available_width = available_width - line_height - Size.span.horizontal_default
```

Comment:
`line_height` comes from the measured `face` `A` 's height.
It's a bit strange to use a height to do substractions on the width :) Is it because the shortcut icon is always a square with a single char ?
Is it (`face`) only used for the shortcut icon ?
I'm a bit confused around here :) may be more explicite names (even if duplicated) could make it clearer.

edit: ok, I see below `item.shortcut_icon_width = line_height -- letter shortcuts of fixed size (1 line)` but it's coming a bit late, after I wrote all this :)